src/client/ua_client.c

```diff
@@ -218,6 +226,10 @@ UA_Client_delete(UA_Client* client) {
         UA_free(client->securityPolicy.certificateVerification);
     }
 
+#ifdef UA_ENABLE_GDS_CLIENT
+    UA_GDS_Client_deinit(client);
```

Comment:
The example secure client (client_basic128rsa15) throws "corrupted double-linked list" error when reaching this point. (Using GDS build configuration in CMakeLists.txt).